pkg/operator/controller/gardenlet/reconciler.go

```diff
@@ -217,3 +225,32 @@ func (r *Reconciler) removeForceRedeployAnnotationIfNeeded(ctx context.Context,
 
 	return nil
 }
+
+func (r *Reconciler) setDefaultGardenClusterAddress(log logr.Logger, gardenlet *seedmanagementv1alpha1.Gardenlet) error {
+	gardenletConfig, err := encoding.DecodeGardenletConfiguration(&gardenlet.Spec.Config, false)
+	if err != nil {
+		return fmt.Errorf("failed to decode gardenlet configuration: %w", err)
+	}
+
+	if gardenletConfig == nil {
+		return nil
+	}
+
+	if gardenletConfig.GardenClientConnection == nil {
+		gardenletConfig.GardenClientConnection = &gardenletconfigv1alpha1.GardenClientConnection{}
+	}
+
+	if gardenletConfig.GardenClientConnection.GardenClusterAddress == nil {
+		log.Info("Setting default garden cluster address", "gardenClusterAddress", r.DefaultGardenClusterAddress)
+		gardenletConfig.GardenClientConnection.GardenClusterAddress = &r.DefaultGardenClusterAddress
+	}
+
+	gardenletConfigRaw, err := encoding.EncodeGardenletConfiguration(gardenletConfig)
+	if err != nil {
+		return fmt.Errorf("failed to encode gardenlet configuration: %w", err)
+	}
+
+	gardenlet.Spec.Config = *gardenletConfigRaw
+
+	return nil
```

Comment:
This is quasi the same code as in the `ManagedSeed` controller. Can you move this out into a function and reuse the majority of it?